### PR TITLE
JVM_IR: Fix field initialization in secondary enum constructors.

### PR DESCRIPTION
**Pull request by madsager**

### CODE REVIEW

File: compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/EnumClassLowering.kt [hunk 1]
```diff
@@ -68,13 +68,42 @@ private class EnumClassLowering(val context: JvmBackendContext) : ClassLoweringP
...
+            insertInstanceInitializer()
             assignOrdinalsToEnumEntries()
             lowerEnumConstructors(irClass)
             lowerEnumEntries()
             setupSynthesizedEnumClassMembers()
             lowerEnumClassBody()
         }
 
+        /* Make sure that an instance initializer call exists, so that field initialization code will be generated. */
+        private fun insertInstanceInitializer() {
```
Comment (madsager) [inline compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/EnumClassLowering.kt]: This code is shared with the JS lowering. Most of the code is different for the lowerings, but we could create a base class and share it between the two.

Comment (udalov) [reply compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/EnumClassLowering.kt]: Yeah, maybe this could be left as a TODO

Comment (madsager) [reply compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/EnumClassLowering.kt]: Done.

**Comment by udalov** [comment]
Thanks!

**Pull request merged**